products/dialogflow/api.yaml

```diff
@@ -129,4 +129,129 @@ objects:
         values:
           - :TIER_STANDARD
           - :TIER_ENTERPRISE
-          - :TIER_ENTERPRISE_PLUS
\ No newline at end of file
+          - :TIER_ENTERPRISE_PLUS
+  - !ruby/object:Api::Resource
+    name: 'Intent'
+    base_url: "projects/{{project}}/agent/intents/"
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    description: |
+      Represents a Dialogflow intent. Intents convert a number of user expressions or patterns into an action. An action
+      is an extraction of a user command or sentence semantics.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/dialogflow/docs/'
+      api: 'https://cloud.google.com/dialogflow/docs/reference/rest/v2/projects.agent.intents'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: |
+          The unique identifier of this intent. 
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        required: true
+        description: |
+          The name of this intent to be displayed on the console.
+      - !ruby/object:Api::Type::Enum
+        name: 'webhookState'
+        description: |
+          Indicates whether webhooks are enabled for the intent.
+          * WEBHOOK_STATE_UNSPECIFIED: Webhook is disabled in the agent and in the intent.
```

Comment:
Usually protobufs require that there is a default for all enums which is "blah_blah_unspecified". This is to guard against bugs where somebody forgot to actually set the value for the enum and is very rarely an acceptable value for the enum. If the API does not accept the unspecified value we usually strip it from the values we present to the user.